internal/envoy/v3/cluster.go

```diff
@@ -58,7 +58,13 @@ func Cluster(c *dag.Cluster) *envoy_cluster_v3.Cluster {
 		cluster.EdsClusterConfig = edsconfig("contour", service)
 	default:
 		// external name set, use hard coded DNS name
-		cluster.ClusterDiscoveryType = ClusterDiscoveryType(envoy_cluster_v3.Cluster_STRICT_DNS)
+		// external name set to LOGICAL_DNS when user selects the ALL loookup family
+		clusterDiscoveryType := ClusterDiscoveryType(envoy_cluster_v3.Cluster_STRICT_DNS)
```

Comment:
could you add tests for the changes in this file

here: https://github.com/projectcontour/contour/blob/ee136791876703db06a9880dbabef2c16ce10ae3/internal/envoy/v3/cluster_test.go#L611

and here: https://github.com/projectcontour/contour/blob/ee136791876703db06a9880dbabef2c16ce10ae3/internal/envoy/v3/cluster_test.go#L223

Reply:
@sunjayBhatia done!